File: Revised/Jav/Stock/StockReducer.java
import java.io.*;
import java.util.*;

public class Reducer {

    public static void main(String[] args) throws IOException {
        String fileName = "output.txt";
        Map<String, Float> maxClosePrices = new HashMap<>();

        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
            String line;
            while ((line = br.readLine()) != null) {
                String[] parts = line.split(" ");

                // Check if the line has at least three parts
                if (parts.length >= 3) {
                    String key = parts[0] + "," + parts[1];
                    float closePrice = Float.parseFloat(parts[2]);

                    if (!maxClosePrices.containsKey(key) || closePrice > maxClosePrices.get(key)) {
                        maxClosePrices.put(key, closePrice);
                    }
		}
            }
        }

        for (Map.Entry<String, Float> entry : maxClosePrices.entrySet()) {
            System.out.println(entry.getKey() + "\t" + entry.getValue());
        }
    }
}

File: Revised/Py/Stock/StockReducer.py
filename = "output.txt"
max_close_prices = {}

with open(filename, 'r') as file:
    for line in file:
        parts = line.strip().split(" ")

        if len(parts) >= 3:
            key = parts[0] + "," + parts[1]
            try:
                close_price = float(parts[2])
            except ValueError:
                # Handle non-float values gracefully
                continue

            if key not in max_close_prices or close_price > max_close_prices[key]:
                max_close_prices[key] = close_price

for key, value in max_close_prices.iteritems():
    print key + "\t" + str(value)
